server/src/routes.ts

```diff
@@ -10,18 +12,94 @@ app.use( (req, res, next) => {
   res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
   next();
 });
-app.use(cors({ origin: ['http://localhost:8080', 'https://donuts-and-drones.fly.dev/'] , methods: ['GET','POST','DELETE','UPDATE','PUT','PATCH'] }));
+app.use(cors({ origin: ['http://localhost:8080'] , methods: ['GET','POST','DELETE','UPDATE','PUT','PATCH'] }));
 app.use(express.json());
 
 app.get('/', (req: Request, res: Response)=>{
   res.status(200).send('Our backend API for EXPLife :)')
 });
 
+/* **************
+ User endpoints
+*************** */
+// get all users
+app.get('/users', async (req: Request, res: Response) => {
+  const allUsers = await User.find();
+  return res.status(200).json(allUsers);
+});
+
+// get specific users
+app.get("/users/:id", async (req: Request, res: Response) => {
+  const { id } = req.params;
+  const user = await User.findById(id);
+  return res.status(200).json(user);
+});
+
+// create new users
+app.post("/users", async (req: Request, res: Response) => {
+  const newUser = new User({ ...req.body });
+  const insertedUser = await newUser.save();
+  return res.status(201).json(insertedUser);
+});
+
+// update specific users
+app.put("/users/:id", async (req: Request, res: Response) => {
+  const { id } = req.params;
+  await User.updateOne({ id }, req.body);
+  const updatedUser = await User.findById(id);
+  return res.status(200).json(updatedUser);
+});
+
+// delete specific users
+app.delete("/users/:id", async (req: Request, res: Response) => {
+  const { id } = req.params;
+  const deletedUser = await User.findByIdAndDelete(id);
+  return res.status(200).json(deletedUser);
+});
+
+/* **************
+ List endpoints
+*************** */
+// get all lists
+app.get('/lists', async (req: Request, res: Response) => {
+  const allLists = await List.find();
+  return res.status(200).json(allLists);
+});
+
+// get specific list
+app.get("/lists/:id", async (req: Request, res: Response) => {
+  const { id } = req.params;
+  const list = await List.findById(id);
+  return res.status(200).json(list);
+});
+
+// create new list
+app.post("/lists", async (req: Request, res: Response) => {
+  const list = new List({ ...req.body });
+  const insertedList = await list.save();
+  return res.status(201).json(insertedList);
+});
+
+// update specific lists
+app.put("/lists/:id", async (req: Request, res: Response) => {
+  const { id } = req.params;
+  await List.updateOne({ id }, req.body);
+  const updatedList = await List.findById(id);
+  return res.status(200).json(updatedList);
+});
+
+// delete specific list
+app.delete("/lists/:id", async (req: Request, res: Response) => {
+  const { id } = req.params;
+  const deletedList = await List.findByIdAndDelete(id);
+  return res.status(200).json(deletedList);
+});
+
```

Comment:
We may also need to have routes for `listitems` so that we can view detailed information, update, and delete individual/specific todo items

Reply:
Yup I can I add those in

Reply:
Good catch. Just updated